### PR TITLE
Picks up item, presses "a" for item count.

### PR DESCRIPTION
**Pull request by jcode89**
You can now pick up an item by walking over it and press "a" to return the number of items in your bag. I had to add in functionality to print the map out after each move, in order to see if I was passing over the items and if they were being added in.

### CODE REVIEW

File: modules/world.py [hunk 1]
```diff
@@ -63,6 +66,21 @@ def exit(self):
             return True
         else:
             return False
+			
+    def pick_up_item(self):
+        key_x = self.data["key"]['x']
+        key_y = self.data["key"]['y']
```
Comment (rubyyot) [inline modules/world.py]: there is a locate function for this.. 

x, y = self.room.locate("key")

File: modules/world.py [hunk 2]
```diff
@@ -63,6 +66,21 @@ def exit(self):
...
+        gold_x = self.data["gold"]['x']
+        gold_y = self.data["gold"]['y']
```
Comment (rubyyot) [inline modules/world.py]: x, y = self.room.locate("gold")

File: modules/world.py [hunk 3]
```diff
@@ -63,6 +66,21 @@ def exit(self):
...
+        x, y = self.locate("player")		
+               
+
```
Comment (rubyyot) [inline modules/world.py]: There is locate(), but what it really looks like you are trying to figure out is what is in the same tile as the player.  Luckily that exists too :)

Room.items(x,y) will tell you everything at those co-ordinates including the player

File: modules/world.py [hunk 4]
```diff
@@ -244,6 +263,10 @@ def exit(self):
         else:
             self.display("Sorry, you cannot exit {0} because you are not at an exit".format(self.room.name))
         return can_exit
+    def item_count(self):
```
Comment (rubyyot) [inline modules/world.py]: Bag.look() has a similar purpose.

File: modules/world.py [hunk 5]
```diff
@@ -274,5 +296,8 @@ def main_loop(self):
                 command_tuple[1]()
             else:
                 self.invalid_command()
+            if self.in_room():
+                self.display(self.room.build_map())
```
Comment (rubyyot) [inline modules/world.py]: I like the map displaying :+1:

**Comment by rubyyot** [comment]
@jcode89 Overall it is awesome seeing this functionality coming together.  Good job.  I left some comments above on places where you could utilize existing code rather than writing new code.  Also there are some failing tests.  I think they have to so with rooms that don't have keys or gold.  You can see the error when you exit level 1 to level 2 (which doesn't have either).  

Really nice to see this all come together, good work!

pr_1

File: modules/world.py (hunk 3 above)
Comment (jcode89) [inline modules/world.py]: I was trying to use locate first and it kept returning an error saying its unable to be hashed. Not totally sure why that happened, but I forgot about Room.items(x, y). Also, you're right the purpose of this method was to check the player location against the item location, if True, add item to bag. 
Jonathan Colón

**Comment by jcode89** [comment]
Okay maybe a way to fix the error is to add a condition like, if player in level 1: continue with item pick up elif player in level 2: Pass? 

Jonathan Colón

**Comment by rubyyot** [comment]
There is a function to return items in room (the keys of the dict) I would
check that for keys and gold before looking for them

**Comment by rubyyot** [comment]
You could do the check in locate() so that it returns negative values or None, None if the item isn't located

**Comment by jcode89** [comment]
Right, right, check to see if the keys of the items dictionary contain key and gold if the do proceed with the code, if not return None. 

Jonathan Colón

**Comment by rubyyot** [comment]
tests pass, that is a good :)

Checking the code....

File: modules/world.py [hunk 6]
```diff
@@ -244,6 +258,20 @@ def exit(self):
         else:
             self.display("Sorry, you cannot exit {0} because you are not at an exit".format(self.room.name))
         return can_exit
+    def item_count(self):
+        key_amount = self.room.bag.how_many("key")
+        gold_amount = self.room.bag.how_many("gold")
+        self.display("You have %d key and %d gold." % (key_amount, gold_amount))
+
+    def map_key(self):
```
Comment (rubyyot) [inline modules/world.py]: I like the map key

**Comment by rubyyot** [comment]
That is looking sweet and.. it is fun to play :)  Well done!  Merge away

**Pull request merged**